src/test/regress/sql/multi_schema_support.sql

```diff
@@ -69,3 +70,301 @@ SELECT master_append_table_to_shard(1190001, 'nation_local', 'localhost', :maste
 
 -- verify table actually appended to shard
 SELECT COUNT(*) FROM "nation._'append";
+
+
+-- test shard creation when search_path is set
+SET search_path TO test_schema_support;
+
+-- create shard with COPY on append distributed table
+CREATE TABLE nation_append_search_path(
+    n_nationkey integer not null,
+    n_name char(25) not null,
+    n_regionkey integer not null,
+    n_comment varchar(152)
+);
+SELECT master_create_distributed_table('nation_append_search_path', 'n_nationkey', 'append');
+
+\COPY nation_append_search_path FROM STDIN with delimiter '|';
+0|ALGERIA|0| haggle. carefully final deposits detect slyly agai
+1|ARGENTINA|1|al foxes promise slyly according to the regular accounts. bold requests alon
+2|BRAZIL|1|y alongside of the pending deposits. carefully special packages are about the ironic forges. slyly special 
+3|CANADA|1|eas hang ironic, silent packages. slyly regular packages are furiously over the tithes. fluffily bold
+4|EGYPT|4|y above the carefully unusual theodolites. final dugouts are quickly across the furiously regular d
+5|ETHIOPIA|0|ven packages wake quickly. regu
+\.
+
+-- create shard with master_create_empty_shard
+SELECT master_create_empty_shard('nation_append_search_path');
+
+-- create shard with master_create_worker_shards
+CREATE TABLE test_schema_support.nation_hash(
+    n_nationkey integer not null,
+    n_name char(25) not null,
+    n_regionkey integer not null,
+    n_comment varchar(152)
+);
+SELECT master_create_distributed_table('test_schema_support.nation_hash', 'n_nationkey', 'hash');
+SELECT master_create_worker_shards('test_schema_support.nation_hash', 4, 1);
+
+
+-- test cursors
+SET search_path TO public;
+BEGIN;
+DECLARE test_cursor CURSOR FOR 
+    SELECT *
+        FROM test_schema_support.nation_append
+        WHERE n_nationkey = 1;
+        FETCH test_cursor;
```

Comment:
Why do you we have  FETCH test_cursor; twice? And why do they return the same result? Do you have a bug in cursors, or do we have the same row twice?

Similarly below